gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/optimization/PathRetractionStrategyTest.java

```diff
@@ -191,6 +200,58 @@ public void doTest() {
                 {__.V().as("a").optional(bothE().dedup().as("b")).
                         choose(select("b"), select("a","b"), project("a").by(select("a"))),
                         "[[[a, b]], [[a, b]], [[a, b]], [[[a, b]]], [[a, b]]]", null},
+                {__.V().as("a").out().select("a").program(new MockVertexProgram(
+                        new HashSet<>(Arrays.asList(TraverserRequirement.LABELED_PATH)))), "[]", null}
         });
     }
-}
+
+    // do nothing mock vertex program for traverser requirement testing
+    private static class MockVertexProgram implements VertexProgram<Tuple> {
```

Comment:
thanks - unless the concrete mock ends up being generally useful to lots of tests or the mockito mock is insanely complicated to the point where no one will understand it, i tend to prefer mockito mocks.